typescript/sdk/src/providers/MultiProvider.ts

```diff
@@ -302,15 +302,17 @@ export class MultiProvider<MetaExt = {}> extends ChainMetadataManager<MetaExt> {
     const contractFactory = await factory.connect(signer);
 
     // estimate gas
-    const deployTx = contractFactory.getDeployTransaction(...params, overrides);
+    const deployTx = contractFactory.getDeployTransaction(...params);
     const gasEstimated = await signer.estimateGas(deployTx);
 
     // deploy with 10% buffer on gas limit
     const contract = await contractFactory.deploy(...params, {
-      ...overrides,
       gasLimit: gasEstimated.add(gasEstimated.div(10)), // 10% buffer
+      ...overrides,
     });
 
+    this.logger.debug({ transaction: deployTx }, `Deploying contract`);
```

Comment:
A few things:
1) msg should come first ('deploying contract')
2) This will make a lot of noise
3) Why wrap the tx object in another object?

Reply:
Should I move it to trace? I do think its useful to be able to plug in the data in tenderly and debug lets say?

Reply:
Yeah good idea

Reply:
Done